assets/js/script.js

```diff
@@ -0,0 +1,181 @@
+      $circle.css("left", "25px") // indicate player2 turn
+
+      if (randomDiceResult === 6) {
+        currentPlayer = $player2
+      } else {
+        currentPlayer = $player1
+      }
+    }
+    gameOver()
+  }
+
+  function gameOver () {
+    if (totalStepsTakenByP1 >= 100 || totalStepsTakenByP2 >= 100) { whoWon() }
+  }
+
+  function whoWon () {
+    if ((totalStepsTakenByP1 >= 100)) {
+      alert('Player 1 has won!')
+      resetButton()
+    } else if ((totalStepsTakenByP2 >= 100)) {
+      alert('Player 2 has won!')
+      resetButton()
+    }
+  }
+
+  function resetButton () {
+    totalStepsTakenByP1 = 1
+    totalStepsTakenByP2 = 1
+    currentPlayer = $player1
+    $player1.detach().appendTo(`#${totalStepsTakenByP1}`)
+    $player2.detach().appendTo(`#${totalStepsTakenByP2}`)
+    $dieValue.text(0)
+    clearInterval(clear)
+  }
+
+  function autoPlayButton() {
+    clear = setInterval(rollDice, 1500)
+  }
+
+  function createTable () {
+    var $tbl = $('#tbl')
+    var id = 100
+    var rowClass = 10
+    // nested loop to create rows and columsn
+    for (var r = 0; r < 10; r++) {
+      var $row = $('<tr>') // create rows
+      $row.attr('class', rowClass--) // give rows an individual class
+
+      for (var c = 0; c < 10; c++) {
+        var $column = $('<td>') // create column
+        $column.css({ 'width': '50px', 'height': '50px'})
+        $column.attr('id', id--) // give each individual column a number
+
+        if(Number($column.attr('id')) % 2 === 0) {
+          $column.css('backgroundColor', '#ffff00')
+        } else {
+          $column.css('backgroundColor', 'white')
+
+        }
+
+        // reverse the numbering for even or odd rows.
+        $row.each(function () {
+          if (Number(($(this).attr('class'))) % 2 === 0) {
+            ($row.append($column))
+          } else {
+            ($row.prepend($column))
+          }
+        })
+        $column.html(id + 1).addClass('cell')
+      }
+      $tbl.append($row)
+    }
+  }
+  createTable()
+  startPosition()
+
+  // Using Array to input Snakes & Ladders
+  var painAndPleasureArray = [0,'ladder0',0,'ladder1',0,0,'snake0',0,'ladder2',0, //row 1
+                            'snake1',0,'ladder3','ladder1',0,0,'snake0','ladder0','snake3','ladder4', //row 2
+                            0,0,0,'snake5',0,0,'ladder3','ladder5','snake1',0 // row 3
+                            ,'ladder2',0,0,'snake2',0,0,0,'ladder4',0,'ladder6' // row 4
+                            ,0,0,'snake4',0,0,0,0,0,0,0  //row 5
+                            ,'ladder7',0,0,'snake2',0,0,0,0,'ladder6',0 //row 6
+                            ,0,'snake3','ladder8','snake4',0,0,'ladder7',0,0,0 // row 7
+                            ,'ladder9',0,'snake6',0,'snake7',0,0,'snake8',0,0 // row 8
+                            ,'ladder8',0,0,'ladder5',0,0,'snake5',0,0,0 //row 9
+                            ,'ladder9',0,'snake6',0,'snake7',0,0,0,'snake8',0] //row 10
+
+
+// Player 1 Snakes & Ladders
```

Comment:
The two functions are looks very similar. How can we merge the functions into one `painOrPlearsure`? Refactor this will be helpful to you.

Hint : you can use parameters for functions.